### PR TITLE
Add multipart markers for event notify 

### PR DESCRIPTION
**Pull request by arempter**

### CODE REVIEW

File: src/main/scala/com/ing/wbaa/rokku/proxy/provider/aws/S3ObjectAction.scala [hunk 1]
```diff
@@ -10,6 +10,12 @@ case class s3ObjectCreated(method: String = "*") extends S3ObjectAction {
 case class s3ObjectRemoved(method: String = "*") extends S3ObjectAction {
   val value = s"s3:ObjectRemoved:$method"
 }
+case class s3MuptipartUpload(method: String = "*") extends S3ObjectAction {
+  val value = s"s3:MultipartUpload:$method"
+}
+case class s3MuptipartUploadComplete(method: String = "*") extends S3ObjectAction {
```
Comment (michallula) [inline src/main/scala/com/ing/wbaa/rokku/proxy/provider/aws/S3ObjectAction.scala]: typo in class name

File: src/main/scala/com/ing/wbaa/rokku/proxy/provider/aws/S3ObjectAction.scala [hunk 2]
```diff
@@ -10,6 +10,12 @@ case class s3ObjectCreated(method: String = "*") extends S3ObjectAction {
 case class s3ObjectRemoved(method: String = "*") extends S3ObjectAction {
   val value = s"s3:ObjectRemoved:$method"
 }
+case class s3MuptipartUpload(method: String = "*") extends S3ObjectAction {
```
Comment (michallula) [inline src/main/scala/com/ing/wbaa/rokku/proxy/provider/aws/S3ObjectAction.scala]: typo in class name

File: src/main/scala/com/ing/wbaa/rokku/proxy/handler/parsers/RequestParser.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+trait RequestParser {
+
+  private def uploadId(queryString: String) = queryString
+    .split("&")
+    .filter(_.contains("uploadId")).flatMap(_.split("=")).toList(1)
+
+  def awsRequestFromRequest(request: HttpRequest): AWSRequest = {
+    val HttpRequest(method, uri, headers, _, _) = request
```
Comment (kr7ysztof) [inline src/main/scala/com/ing/wbaa/rokku/proxy/handler/parsers/RequestParser.scala]: unused headers
```suggestion
    val HttpRequest(method, uri, _, _, _) = request
```

File: src/main/scala/com/ing/wbaa/rokku/proxy/handler/parsers/RequestParser.scala [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    val queryString = uri.queryString().getOrElse("")
+    val containsUploadId = queryString.contains("uploadId")
+    val isXML = request.entity.contentType.mediaType == MediaTypes.`application/xml`
+    val isOctetStream = request.entity.contentType.mediaType == MediaTypes.`application/octet-stream`
+
+    method match {
+      // aws multipart part upload eg. PUT /ObjectName?uploadId=UploadId
+      case HttpMethods.PUT if containsUploadId => AWSRequest(MultipartRequest(uploadId(queryString), false))
```
Comment (kr7ysztof) [inline src/main/scala/com/ing/wbaa/rokku/proxy/handler/parsers/RequestParser.scala]: Hint from Intellij - for readability
```suggestion
      case HttpMethods.PUT if containsUploadId => AWSRequest(MultipartRequest(uploadId(queryString), completeMultipartUpload = false))
```

File: src/main/scala/com/ing/wbaa/rokku/proxy/handler/parsers/RequestParser.scala [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      // aws multipart complete eg. POST /ObjectName?uploadId=UploadId and content-type application/xml
+      case HttpMethods.POST if containsUploadId && (isXML || isOctetStream) => AWSRequest(MultipartRequest(uploadId(queryString), true))
```
Comment (kr7ysztof) [inline src/main/scala/com/ing/wbaa/rokku/proxy/handler/parsers/RequestParser.scala]:
```suggestion
      case HttpMethods.POST if containsUploadId && (isXML || isOctetStream) => AWSRequest(MultipartRequest(uploadId(queryString), completeMultipartUpload = true))
```

File: src/main/scala/com/ing/wbaa/rokku/proxy/handler/parsers/RequestParser.scala [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+object RequestParser {
+
+  trait RecognizedRequest
+
+  case class RequestUnknown() extends RecognizedRequest
+
+  case class MultipartRequest(uploadId: String, completeMultipartUpload: Boolean) extends RecognizedRequest
+
+  case class AWSRequest(recognizedRequest: RecognizedRequest)
```
Comment (kr7ysztof) [inline src/main/scala/com/ing/wbaa/rokku/proxy/handler/parsers/RequestParser.scala]: not needed

**Pull request merged**